File: app/src/main/java/org/pocketx/knell/utils/Injector.java
package org.pocketx.knell.utils;

import android.annotation.SuppressLint;
import android.app.Application;

import org.pocketx.knell.domain.BirthdayManager;

/**
 * Created by zhangchao on 2017/10/31.
 */

public final class Injector {

    private static final String BIRTHDAY_SERVICE = "org.pocketx.knell.birthday";

    public static boolean matchService(String name){
        return BIRTHDAY_SERVICE.equals(name);
    }

    @SuppressLint("WrongConstant")
    public static BirthdayManager obtain(Application application){
        return (BirthdayManager) application.getSystemService(BIRTHDAY_SERVICE);
    }

    private Injector(){
        throw new AssertionError("No instance.");
    }

}

File: app/src/main/java/org/pocketx/knell/utils/Injector.kt
package org.pocketx.knell.utils

import android.annotation.SuppressLint
import android.app.Application

import org.pocketx.knell.domain.BirthdayManager

/**
 *
 * @author Shadow
 * @author shenghaiyang
 */
class Injector private constructor() {

    companion object {

        private const val BIRTHDAY_SERVICE = "org.pocketx.knell.birthday"

        fun matchService(name: String): Boolean {
            return BIRTHDAY_SERVICE == name
        }

        @SuppressLint("WrongConstant")
        fun obtain(application: Application): BirthdayManager {
            return application.getSystemService(BIRTHDAY_SERVICE) as BirthdayManager
        }
    }

}
